client/js/components/tags/TagsListForm.vue

```diff
@@ -0,0 +1,262 @@
+        @click="() => openForm('topic')" />
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapActions, mapMutations, mapState } from 'vuex'
+import AddonWrapper from '@DpJs/components/addon/AddonWrapper'
+import {
+  DpButton,
+  DpButtonRow,
+  DpInput,
+  DpLabel,
+  DpSelect,
+  dpValidateMixin
+} from '@demos-europe/demosplan-ui'
+
+export default {
+  name: 'TagsListForm',
+
+  components: {
+    AddonWrapper,
+    DpButton,
+    DpButtonRow,
+    DpInput,
+    DpLabel,
+    DpSelect,
+  },
+
+  mixins: [dpValidateMixin],
+
+  props: {
+    procedureId: {
+      type: String,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      currentForm: '',
+      newTag: {
+        title: '',
+        topic: ''
+      },
+      newTopic: {
+        title: ''
+      },
+      // This is necessary to allow extending the Tags-Resource
+      tagAttributes: {
+        boilerplate: '',
+        title: ''
+      }
+    }
+  },
+
+  computed: {
+    ...mapState('TagTopic', {
+      TagTopic: 'items'
+    }),
+
+    topicsAsOptions () {
+      return Object.values(this.TagTopic).map(category => {
+        const { attributes, id } = category
+
+        return {
+          label: attributes.title,
+          value: id
+        }
+      })
+    }
+  },
+
+  methods: {
+    ...mapMutations('TagTopic', {
+      updateTagTopic: 'setItem'
+    }),
+
+    ...mapActions('Tag', {
+      createTag: 'create'
+    }),
+
+    ...mapActions('TagTopic', {
+      createTagTopic: 'create'
+    }),
+
+    closeForm () {
+      this.currentForm = ''
+      this.resetFormData()
+    },
+
+    openForm (form) {
+      this.currentForm = form
+    },
+
+    resetFormData () {
+      this.newTopic.title = ''
+      this.newTag = {
+        title: '',
+        topic: ''
+      }
+    },
+
+    saveNewTag () {
+      this.createTag({
+        type: 'Tag',
+        attributes: {
+          title: this.newTag.title
+        },
+        relationships: {
+          topic: {
+            data: {
+              type: 'TagTopic',
+              id: this.newTag.topic
+            }
+          }
+        }
+      })
+        .then(response => {
+          if (!response.data.Tag || !this.TagTopic[this.newTag.topic]) {
+            return
+          }
+
+          const parentTopic = this.TagTopic[this.newTag.topic]
+          const newTagId = Object.keys(response.data.Tag)[0]
+
+          this.updateTagTopic({
+            id: this.newTag.topic,
+            type: 'TagTopic',
+            attributes: parentTopic.attributes,
+            relationships: {
+              ...parentTopic.relationships,
+              tags: {
+                data: parentTopic.relationships.tags.data.concat({
+                  type: 'Tag',
+                  id: newTagId
+                })
+              }
+            }
+          })
+
+          this.$root.$emit('tag:created', newTagId)
+          // reset form data
+          this.closeForm()
+          this.newTag = this.tagAttributes
```

Comment:
this is done in `closeForm` inherently

Reply:
https://github.com/demos-europe/demosplan-core/pull/4256/commits/a66f487c978895bde9aaa4336418b9f8781d59ec